test/dates/periods.jl

```diff
@@ -292,6 +322,15 @@ dt = Dates.DateTime(2014)
 @test Dates.toms(d)  == Dates.value(Dates.Millisecond(d)) == 86400000
 @test Dates.toms(w)  == Dates.value(Dates.Millisecond(w)) == 604800000
 
+@test Dates.tons(ns) == Dates.value(Dates.Nanosecond(ns)) == 1
+@test Dates.tons(us) == Dates.value(Dates.Nanosecond(us)) == 1000
+@test Dates.tons(ms) == Dates.value(Dates.Nanosecond(ms)) == 1000000
+@test Dates.tons(s)  == Dates.value(Dates.Nanosecond(s))  == 1000000000
+@test Dates.tons(mi) == Dates.value(Dates.Nanosecond(mi)) == Int64(60000000000)
```

Comment:
Yes, happening on both. I'll see if I can't get a 32-bit build on my windows machine to test it out.